cdc/log.cc

```diff
@@ -885,11 +899,22 @@ class transformer final {
             });
         }
         if (!p.clustered_rows().empty()) {
-            p.clustered_rows().begin()->row().cells().for_each_cell([&] (column_id id, const atomic_cell_or_collection&) {
-                auto& cdef =_schema->column_at(column_kind::regular_column, id);
-                regular_columns.emplace_back(id);
-                columns.emplace_back(&cdef);
+            bool has_row_delete = std::any_of(p.clustered_rows().begin(), p.clustered_rows().end(), [] (const rows_entry& re) {
+                return re.row().deleted_at();
             });
+
+            if (has_row_delete) {
+                for (const column_definition& c: _schema->regular_columns()) {
+                    regular_columns.emplace_back(c.id);
+                    columns.emplace_back(&c);
+                }
+            } else {
+                p.clustered_rows().begin()->row().cells().for_each_cell([&] (column_id id, const atomic_cell_or_collection&) {
+                    auto& cdef =_schema->column_at(column_kind::regular_column, id);
```

Comment:
while we're at it, a const here wouldn't hurt either

Reply:
this is const& already because schema::column_at returns const&